onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -63,8 +63,10 @@ def aten_adaptive_max_pool1d(
 def aten_add(self, other, alpha: float = 1) -> TensorType:
     # add.Tensor(Tensor self, Tensor other, *, Scalar alpha=1) -> Tensor
     if alpha != 1:
-        other = op.Mul(other, alpha)  # type: ignore[arg-type]
-    return op.Add(self, other)
+        other_scaled = op.Mul(other, alpha)  # type: ignore[arg-type]
+    else:
+        other_scaled = other
```

Comment:
Such logic is needless. This will cause a different behavior than the PyTorch aten op.
In PyTorch torch.add() method, if the alpha is 0, then the returned value is just self.

Reply:
More importantly, I wonder how much we want alpha to appear here, as in most cases it’s just 1, but the generated function will always have a conditional inserted. Will that hurt performance? Do we need a mechanism to specialize the functions in the optimization phase? Can we leave this to optimization, or should we dispatch to a specialized version already during conversion?

Reply:
Conditionals that depend only on attribute-values can always be optimized away. We will need such optimizations in onnxruntime, and we do plan to add it in onnxruntime / onnx (extending the existing constant-folding optimization). In short, we need it, but it would be better if we don't duplicate this effort.

Reply:
> onnx complained that the previous logic needs to be SSA. I think this is a gap be need to bridge #242 or find a better solution for

Per my understanding, to resolve SSA issue, we just need to keep the line 66 as the latest one. But the logic of line 67-68 doesn't have a relationship to SSA issue, right?

Reply:
I don't understand the SSA issue. While ONNX does not allow multiple assignments to the same variable, the onnx script converter is supposed to take care of this and rename variables as needed. If there is a problem, there must be a bug in the converter. Eg., see https://github.com/microsoft/onnx-script/blob/main/onnxscript/test/if_test.py ...

Reply:
Awesome. so it sounds like we can keep the functions as complete as possible and leave the simplification to optimizers. As long as the optimization doesn't only exist in onnxruntime.

Reply:
https://github.com/microsoft/onnx-script/pull/244 should fix the SSA issue, hopefully.